presto-main/src/main/java/com/facebook/presto/operator/window/PagesWindowIndex.java

```diff
@@ -80,6 +82,14 @@ public Slice getSlice(int channel, int position)
     }
 
     @Override
+    public Block getBlock(int channel, int position)
+    {
+        BlockBuilder builder = pagesIndex.getType(channel).createBlockBuilder(new BlockBuilderStatus(), 1);
```

Comment:
I talked to @dain about this. There is one question to answer: do you want a view, which holds onto all the memory of the original block, or do you want a copy?

For normal window functions, a view seems fine, since the entire thing is in memory. For streaming, this would be a problem, as the accumulator could end up referencing the entire input. Thus, we want a copy.

The right thing is to call `getSingleValueBlock()` which can have an efficient implementation. If we wanted view, `getRegion()` is the correct call.